src/Microsoft.DotNet.Build.Tasks/PackageFiles/versioning.targets

```diff
@@ -230,7 +229,22 @@
       <OldBuildVersionFiles Include="$(BuildVersionFilePath)BuildVersions-*.props" Exclude="$(BuildVersionFilePath)%(BuildVersionFileItem.Filename).props" />
     </ItemGroup>
     <Delete Files="@(OldBuildVersionFiles)" TreatErrorsAsWarnings="true"/>
+  </Target>
+
+  <Target Name="GenerateBuiltByString" Condition="'$(BuiltByString)'==''">
+
+    <!-- If building on Windows, default BuiltByString to use Computername and username -->
+    <BuiltByString Condition="'$(OSEnvironment)'=='Windows_NT'">%20built by: $(COMPUTERNAME)-$(USERNAME)</BuiltByString>
```

Comment:
Almost what I had in mind :)

To enable folks to setup an environment so they don't have to run this whoami/hostname commands we can allow them to set COMUTERMAME and USERNAME env variables. So I would default VersionUserName and VersionHostName to those guys and only execute whoami/hostname if we are not on Windows and if the particular property is empty.